### PR TITLE
Add different icons for different component types

### LINKED ISSUE #750: Represent the component with icons based on the component type
This issue will help in better representation of icons for the components. Currently there are 3 different ways to create a component, so the UI representation will also cater to display 3 different icons for 3 different components (git, binary, local folder)

### PR DESCRIPTION
**Pull request by mohitsuman**
Fixes #750 

- This also fixes the icons for (#753)

   - projects
   - applications
   - storage
   - cluster url
   - login
   - refresh

Now the icons are more robust and clear on different themes of VSCode. This is the UI change which will happen. 

Screenshot 2019-04-02 at 3 39 13 AM

### CODE REVIEW

File: src/odo.ts [hunk 1]
```diff
@@ -361,14 +373,25 @@ export class OdoImpl implements Odo {
...
             // see https://github.com/openshift/odo/issues/1521
         }
-        const apps: string[] = data.map((value) => value.metadata.name);
-        return apps.map<OpenShiftObject>((value) => new OpenShiftObjectImpl(application, value, ContextType.COMPONENT, this, TreeItemCollapsibleState.Collapsed));
+        const apps = data.map(value => ({ name: value.metadata.name, source: value.spec.source }));
+        return apps.map<OpenShiftObject>((value) =>
+            {
+                if (value.source.startsWith('https://')) {
+                    compSource = 'git';
+                } else if (value.source.endsWith('.jar|')) {
```
Comment (dgolovin) [inline src/odo.ts]: binary components can be created with any file, not only jar ones.

File: src/odo.ts [hunk 2]
```diff
@@ -361,14 +373,25 @@ export class OdoImpl implements Odo {
     public async _getComponents(application: OpenShiftObject): Promise<OpenShiftObject[]> {
         const result: cliInstance.CliExitData = await this.execute(Command.listComponents(application.getParent().getName(), application.getName()));
         let data: any[] = [];
+        let compSource = '';
```
Comment (dgolovin) [inline src/odo.ts]: Should be  declared inside array.map callback on line 384.`

File: src/odo.ts (hunk 1 above)
Comment (mohitsuman) [reply src/odo.ts]: yeah I need to add the condition to test all the probabilities. Need to figure out a way to get them using a regex or something.

Comment (mohitsuman) [reply src/odo.ts]: Updated with the fix.

File: src/odo.ts (hunk 2 above)
Comment (mohitsuman) [reply src/odo.ts]: Fixed.

File: src/odo.ts [hunk 3]
```diff
@@ -367,8 +380,20 @@ export class OdoImpl implements Odo {
...
-        const apps: string[] = data.map((value) => value.metadata.name);
-        return apps.map<OpenShiftObject>((value) => new OpenShiftObjectImpl(application, value, ContextType.COMPONENT, this, TreeItemCollapsibleState.Collapsed));
+        const componentObject = data.map(value => ({ name: value.metadata.name, source: value.spec.source }));
+
+        return componentObject.map<OpenShiftObject>((value) =>
+            {
+                let compSource: string = '';
+                if (value.source.startsWith('https://')) {
+                    compSource = 'git';
+                } else if (Archive.endsWithAny([".tar", ".jar", ".iso", ".tar.gz", ".zip"], value.source)) {
```
Comment (dgolovin) [inline src/odo.ts]: We cannot just restrict to files with extensions for archives. There are many others out there: war, ear and etc.  It should check if ```value.source``` is path to the existing file.

**Comment by dgolovin** [comment]
@mohitsuman Icon for Service is still not updated.

**Comment by mohitsuman** [comment]
> @mohitsuman Icon for Service is still not updated.

Fixed with the updated icon.

**Comment by mohitsuman** [comment]
testPR

**Comment by mohitsuman** [comment]
@dgolovin Does this look good to you ?

**Comment by mohitsuman** [comment]
testPR

**Pull request merged**